tools/crm_simulate.c

```diff
@@ -73,6 +76,274 @@ extern gboolean bringing_nodes_online;
 	}					\
     } while(0)
 
+#define INDENT "                                   "
+
+static gboolean
+in_place_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error) {
+    options.store = TRUE;
+    options.process = TRUE;
+    options.simulate = TRUE;
+    return TRUE;
+}
+
+static gboolean
+live_check_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error) {
+    if (options.xml_file) {
+        free(options.xml_file);
+    }
+
+    options.xml_file = NULL;
+    return TRUE;
+}
+
+static gboolean
+node_down_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error) {
+    options.modified++;
+    options.node_down = g_list_append(options.node_down, (gchar *) g_strdup(optarg));
```

Comment:
Now that we're strdup'ing these items, we also need to free them.

I'm wondering how it ever worked before (without strdup).

Reply:
That's a good question that I do not have the answer to.  Looking around for several minutes doesn't show anything obvious.  Anyway, I've added a giant block of freeing after the done label.